addons/virtualcamera/VCameras/VCameraTrigger.gd

```diff
@@ -0,0 +1,24 @@
+extends Area
+
+class_name VCameraTrigger
+
+export(NodePath) var target_vcamera : NodePath
+var overlapping_count : int = 0 setget set_overlapping_count
```

Comment:
Maybe this should be "private" by prefixing with an underscore (eg: `_overlapping_count`)? I can see potential misuse by allowing other scripts to modify this directly.

I'm not going to force this change, just wanted to make you aware of it and give you the chance to make the decision.

Reply:
Good idea! Gdscript style guide recommends adding underscore to should-be-private variables, but I haven't picked up this habit yet.

I also considered having "invert" bool, but I had hard time coming up with actual use case. I'd imagine the workflow as follows: set up VCamera priorities in VCameras themselves. Most usual scenario: normal player camera has small priority, other viewpoints have higher priority but are disabled until player steps into trigger. Setting priorities from multiple places leads easily into situation, where dev forgets some places where they get set and if there needs to be changes later on, they get unexpected behaviour. Same kind of problems that you get from tightly coupled code.

Anyway this is my take on this, feel free to give more insight how these different variants could be used.

Reply:
After giving it more thought, dropdown menu sounds the right solution. Flexibility is nice, as long default parameters do the most sensible thing. Changing VCamera groups is another option which could be there. I'm not yet sure how to accomplish this in elegant manner, but there should be a way to have camera join one group if first player triggers it, and second group if second player triggers it.

However, changing the priority comes with extra challenge. What should the trigger do after target leaves? Remember original priority and return it? Or give second parameter which defines what should priority be set afterwards?

On note of flexibility, modifying multiple VCameras at once could be something the user wants.

All this leads me to think about separate `VCameraAction` resource, which can be added as an action to trigger. `VCameraTrigger` would have an array `actions` where user can add as many actions they need, be they enabling, disabling or adjusting priority or groups. Minus side of this is tad poor support from the editor (see godotengine/godot-proposals#18).

Reply:
I tested having actions as a resource, but it felt cumbersome. Found better solution instead!